internal/config/tls_credentials.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+// TLSCredentials holds the fields for TLS credentials
+// that are used for starting a server.
+type TLSCredentials struct {
+	// CertFile is the file path containing the TLS certificate.
+	CertFile string `yaml:"cert_file"`
```

Comment:
add mapstructure annotation: iirc viper requires mapstructure so I don't think that this will work for the collector...

Reply:
I believe we are also mixing `yaml` and `mapstructure` which we shouldn't do -- let's use `mapstructure` everywhere

Reply:
I thought about that while making this PR, but alas I noticed that inside `internal/config` we use `yaml` https://github.com/census-instrumentation/opencensus-service/blob/9f8f610cebd4e1bcc93a306786af527614dde11d/internal/config/config.go and when parsing by Viper, we explicitly set the type to "yaml" in https://github.com/census-instrumentation/opencensus-service/blob/9f8f610cebd4e1bcc93a306786af527614dde11d/internal/config/viperutils/viper.go#L9-L18
so this should work